haystack/dataclasses/chat_message.py

```diff
@@ -59,6 +59,31 @@ class ToolCall:
     arguments: Dict[str, Any]
     id: Optional[str] = None  # noqa: A003
 
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert ToolCall into a dictionary.
+
+        :returns: A dictionary with keys 'tool_name', 'arguments', and 'id'.
+        """
+        return {"tool_name": self.tool_name, "arguments": self.arguments, "id": self.id}
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "ToolCall":
+        """
+        Creates a new ToolCall object from a dictionary.
+
+        :param data:
+            The dictionary to build the ToolCall object.
+        :returns:
+            The created object.
+        """
+        if "tool_name" not in data or "arguments" not in data:
+            raise ValueError(
+                "Fields `tool_name` and `arguments` are required for ToolCall deserialization. "
+                f"Received dictionary with keys: {list(data.keys())}"
+            )
+        return ToolCall(tool_name=data["tool_name"], arguments=data["arguments"], id=data.get("id"))
```

Comment:
I think we can simplify this to the below. This way we rely on the builtin checking by dataclass instead of raising our own `ValueError`
```suggestion
        return ToolCall(**data)
```